components/ILIAS/Notifications/classes/Setup/ilNotificationsSetupAgent.php

```diff
@@ -33,7 +33,7 @@ class ilNotificationsSetupAgent implements Agent
 {
     use HasNoNamedObjective;
 
-    public function __construct(protected Factory $refinery)
+    public function __construct(protected readonly Factory $refinery, protected readonly array $provider)
```

Comment:
Please define the shape of `$provider`, this should be a list of providers: `list<\ILIAS\Notifications\Interfaces\PushProviderInterface>` I guess.